TESTING.md

```diff
@@ -0,0 +1,145 @@
+[envtest]: https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest
+
+### Integration tests
+
+KIC's integration tests rely on its controller manager being run in the same
+process as the tests via [`manager.Run()`][manager_run].
+
+These tests rely on a [test suite setup][integration_test_suite]
+using [`TestMain()`][pkggodev_testmain].
+
+Said setup will either create a new [Kind](https://kind.sigs.k8s.io) cluster or use an existing one to run the
+tests against.
+
+Most of the setup - and cleanup after the tests have run - is being done using
+[ktf][ktf].
+
+Currently, the cluster is being shared across all the tests that are a part
+of the test suite so special care needs to be taken in order to clean up after the tests
+that run.
+The typical approach is to run them in a dedicated, disposable Kubernetes namespace created just for the purposes of that test.
+
+#### How to run
+
+Integration tests are located under `tests/integration/` and use `integration_tests`
+[build tag][go_build_tag].
+
+You can run them using one of the dedicated Makefile targets:
+
+- `test.integration` run all integration tests with standard verbose output on stderr.
+  This will run tests for dbless, postgres, enterprise and non enterprise.
+
+- `test.integration.dbless` run all dbless integration tests with standard
+  verbose output on stderr. The output will also include controllers' logs.
+
+- `test.integration.postgres` run all postgres integration tests with standard
+  verbose output on stderr. The output will also include controllers' logs.
+
+Through `GOTESTFLAGS` you can specify custom flags that will be passed to `go test`.
+This can allow you to run a subset of all the tests for faster feedback times, e.g.:
+
+```
+make test.integration.dbless GOTESTFLAGS="-count 1 -run TestUDPRouteEssentials"
+```
+
+[ktf]: https://github.com/Kong/kubernetes-testing-framework
+[pkggodev_testmain]: https://pkg.go.dev/testing#hdr-Main
+[integration_test_suite]: https://github.com/Kong/kubernetes-ingress-controller/blob/61e06ee64ff913aa9952816121125fca7ed59ba5/test/integration/suite_test.go#L36
+[manager_run]: https://github.com/Kong/kubernetes-ingress-controller/blob/5abc699aeee552945a76c82e3f7abb3e1b2fabf1/internal/cmd/rootcmd/run.go#L14-L22
+
+### End-to-End (E2E) tests
+
+End to end tests in KIC are used to test features or use cases which rely on KIC
+being deployed in-cluster.
+
+For instance:
+
+- deploying all in one manifests tested in [all in one tests][test_e2e_all_in_one]
+- upgrade scenarios tested in [upgrade tests][test_e2e_upgrade]
+
+These tests deploy KIC and Kong Gateway in a cluster using the requested image(s)
+which could be customized via dedicated environment variables like:
+
+- [`TEST_KONG_CONTROLLER_IMAGE_OVERRIDE`][env_var_controller_image_override]
+- [`TEST_KONG_IMAGE_OVERRIDE`][env_var_kong_image_override]
+
+On CI, those are being run both in kind and GKE environments.
```

Comment:
```suggestion
On CI, those are being run both in Kind and GKE environments.
```
nit

Reply:
all mentions of kind at https://kind.sigs.k8s.io/ use kind not Kind. 🤔

Reply:
oh my bad, so in `TESTING.md:64` I suggested wrongly `Kind` and should be `kind` too 🤣